File: LeetCode/933.py
#Link: https://leetcode.com/problems/number-of-recent-calls/

class RecentCounter:
    recent = list()

    def __init__(self):
        self.recent = []

    def ping(self, t: int) -> int:
        while self.recent and self.recent[0] + 3000 < t:
            self.recent.pop(0)
        self.recent.append(t)
        return len(self.recent)


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: LeetCode/cpp/933.cpp
//Link: https://leetcode.com/problems/number-of-recent-calls/

class RecentCounter {
public:
    queue<int> recent;
    
    RecentCounter() {
    }
    
    int ping(int t) {
        while(!recent.empty() && recent.front() + 3000 < t)
            recent.pop();
        recent.push(t);
        return recent.size();
    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */
 